### PR TITLE
MGMT-2234 Add subsystem test for Auth

### PR DESCRIPTION
**Pull request by rollandf**
Test with Agent with SSO failures

### CODE REVIEW

**Comment by rollandf** [comment]
@danielerez @filanov

File: subsystem/host_test.go [hunk 1]
```diff
@@ -477,8 +477,31 @@ var _ = Describe("Host tests", func() {
...
 		})
 		Expect(err).To(HaveOccurred())
+		Expect(err).Should(BeAssignableToTypeOf(installer.NewRegisterHostUnauthorized()))
 
 		err = wiremock.DeleteStub(wrongTokenStubID)
 		Expect(err).ToNot(HaveOccurred())
 	})
+
+	It("register_failed_sso", func() {
+		if !Options.EnableAuth {
```
Comment (empovit) [inline subsystem/host_test.go:487]: What's the purpose? Does it mean that this functionality won't be tested if auth is disabled? I'd probably write two tests then - one with auth disabled and the other with forced enabled.

File: subsystem/subsystem_suite_test.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ import (
 )
 
 var db *gorm.DB
-var agentBMClient, badAgentBMClient, userBMClient, adminUserBMClient, unallowedUserBMClient *client.AssistedInstall
+var agentBMClient, badAgentBMClient, anotherAgentBMClient, userBMClient, adminUserBMClient, unallowedUserBMClient *client.AssistedInstall
```
Comment (empovit) [inline subsystem/subsystem_suite_test.go:21]: What does "another" mean in this context? I couldn't figure this one, the others are more or less clear (maybe except the "bad").

File: subsystem/wiremock_stubs.go [hunk 3]
```diff
@@ -47,6 +47,7 @@ const (
 	fakePayloadAdmin         string = "admin@example.com"
 	fakePayloadUnallowedUser string = "unallowed@example.com"
 	FakePS                   string = "dXNlcjpwYXNzd29yZAo="
+	AnotherFakePS            string = "dXNlcjI6cGFzc3dvcmQyCg=="
```
Comment (empovit) [inline subsystem/wiremock_stubs.go:50]: Does "PS" stand for "pull secret"?

File: subsystem/host_test.go (hunk 1 above)
Comment (rollandf) [reply subsystem/host_test.go:487]: RegisterHost is covered in multiple tests with `registerHost` https://github.com/openshift/assisted-service/blob/master/subsystem/utils_test.go#L29
This specific test is relevant only when Auth is enable, which is by the way the default in sub-system tests unless running `on-prem`.

File: subsystem/subsystem_suite_test.go (hunk 2 above)
Comment (rollandf) [reply subsystem/subsystem_suite_test.go:21]: I need to use a different `Agent` client to not interfere with other tests.
Any suggestion for a better name?

File: subsystem/wiremock_stubs.go (hunk 3 above)
Comment (rollandf) [reply subsystem/wiremock_stubs.go:50]: Yes

File: subsystem/subsystem_suite_test.go (hunk 2 above)
Comment (empovit) [reply subsystem/subsystem_suite_test.go:21]: by the scenario you're testing. e.g. failedSSOAgent

File: subsystem/wiremock_stubs.go (hunk 3 above)
Comment (empovit) [reply subsystem/wiremock_stubs.go:50]: How about failedSSOFakePullSecret?

Comment (rollandf) [reply subsystem/wiremock_stubs.go]: Done

File: subsystem/subsystem_suite_test.go (hunk 2 above)
Comment (rollandf) [reply subsystem/subsystem_suite_test.go]: Done

**Review by ronniel1 — APPROVED** [review]
/approve

**Comment by rollandf** [comment]
/unhold
Rebased PR

**Comment by rollandf** [comment]
Closing for now, as OCM connection holds a cache of the SSO token, making this test fail.

**Comment by filanov** [comment]
> Closing for now, as OCM connection holds a cache of the SSO token, making this test fail.

@rollandf We have a specific deployment for subsystem for example monitor interval is just 1 second, maybe you can set a shorter cache configuration as well.

**Comment by rollandf** [comment]
> > Closing for now, as OCM connection holds a cache of the SSO token, making this test fail.
> 
> @rollandf We have a specific deployment for subsystem for example monitor interval is just 1 second, maybe you can set a shorter cache configuration as well.

@filanov 
The caching is in the OCM SDK, it checks if the current token is about to expire.

**Pull request closed without merging**